src/edit/useFormScoresEdit.js

```diff
@@ -0,0 +1,179 @@
+import { useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import {
+    validateScoreIsInRange
+} from '../app/validators.services'
+import loadLocally from '../lib/loadLocally'
+import removeLocally from '../lib/removeLocally'
+import saveLocally from '../lib/saveLocally'
+
+const STORAGE_KEY = 'inputsScoresEdit'
+
+export default function useFormScoresEdit({
+    targetProfile,
+    editGameProfile,
+    updateVisibleForm }) {
+    
+    const history = useHistory()
+
+    const [inputsScores, setInputsScores] = useState(loadLocally(STORAGE_KEY) ?? targetProfile)
+    
+    const validInputs = inputsScores.playersArray && prepareInputValidation(inputsScores)
+    const hasAllValidInputs = validInputs && condenseValidInputs()
+
+    const isSaveButtonShownScores = hasAllValidInputs
+    
+    const [dirtyInputs, setDirtyInputs] = useState({})
+    const hasAllDirtyInputs = dirtyInputs && condenseDirtyInputs()
+
+    useEffect(() => saveLocally(STORAGE_KEY, inputsScores), [inputsScores])
+    useEffect(() => setInputsScores(targetProfile), [targetProfile])
+    useEffect(() => prepareInputValidation(targetProfile), [targetProfile])
+    useEffect(() => prepareDirtyInputs(targetProfile), [targetProfile])
```

Comment:
I think you can also combine the useEffects here.

Reply:
Same here :)